flax/nnx/summary.py

```diff
@@ -206,13 +183,61 @@ def _create_obj_env(object_types):
         result[(obj_type, name)] = top_method
   return result
 
-def _argsave(tracer_args, f):
+def _get_inputs_repr(args, kwargs):
+  input_args, input_kwargs = jax.tree.map(
+    _to_dummy_array, (args, kwargs)
+  )
+  inputs_repr = ''
+  if input_args:
+    if len(input_args) == 1 and not input_kwargs:
+      inputs_repr += _as_yaml_str(input_args[0])
+    else:
+      inputs_repr += _as_yaml_str(input_args)
+    if input_kwargs:
+      inputs_repr += '\n'
+  if input_kwargs:
+    inputs_repr += _as_yaml_str(input_kwargs)
+  return inputs_repr
+
+def _save_call_info(counter, tracer_args, f, node_stats, compute_flops, compute_vjp_flops):
   "Wrap a function to save its arguments"
-  n = f.__name__
+
+  # Used when computing vjp flops
+  def do_vjp(*args, **kwargs):
+    primals, f_vjp = jax.vjp(f, *args, **kwargs)
+    return f_vjp(primals)
+
+  method_name = f.__name__
+
+  @functools.partial(jax.jit)
+  def jit_f(graphdef, state):
+    args, kwargs = nnx.merge(graphdef, state)
+    return f(*args, **kwargs)
+  jit_f._seen = set()
```

Comment:
Q: is it necessary to attach `_seen` as an attribute? wouldn't it be the same to have a as a separate variable?

Reply:
Good point. It would be better to have it as a separate variable. Let me add that.